src/transactions/PayForFiles.ts

```diff
@@ -0,0 +1,28 @@
+import { Transaction } from './Transaction';
+import { Keypair, PublicKey, SystemProgram, TransactionInstruction } from '@solana/web3.js';
+import { MemoProgram } from './programs/MemoProgram';
+
+export class PayForFiles extends Transaction {
+  constructor(
+    feePayer: PublicKey,
+    lamports: number,
+    filesHash: Buffer,
+    arweaveWallet = new PublicKey('HvwC9QSAzvGXhhVrgPmauVwFWcYZhne3hVot9EbHuFTm'),
```

Comment:
Transactions should know the absolute possible minimum about the way they're generated, so I'm decoupling all the stuff that happens to produce the filesHash and lamports and giving that job to the user. (to be later done in helper convenience utility)

Reply:
Also `arweaveWallet` should probably go to a separate constants folder but I'm waiting to see how many of these exist before making that decision.